stdlib/public/core/Integers.swift.gyb

```diff
@@ -1661,22 +1661,145 @@ ${assignmentOperatorComment(x.nonMaskingOperator, False)}
+  /// Thus, we need to choose a rounding rule that describes how to round
+  /// the quotient when it is not an exact integer. This function provides
+  /// the ability to control the rounding.
+  ///
+  /// A few of the available rounding rules are worth calling out in particular:
+  ///
+  /// - `.towardZero` matches the behavior of the `/` operator.
+  ///
+  /// - `.down` is frequently called "floored" or "flooring" division.
+  ///
+  /// - `.up` and `.awayFromZero` are useful when working with sizes and counts.
+  ///   If you want to put `n` objects into buckets, each of which can hold `k`
+  ///   objects, you will use `n.divided(by: k, rounding: .up)` buckets.
+  ///
+  /// - `.toNearestOrEven` and `.toNearestOrAwayFromZero` are useful if you
+  ///   need to emulate or implement floating-point rounding with integer
+  ///   arithmetic.
+  ///
+  /// - Parameters:
+  ///   - divisor: the number to divide by.
+  ///   - rule: the rounding rule to use.
+  ///
+  /// See also `remainder(dividingBy:, roundingQuotient:)` and
+  /// `quotientAndRemainder(dividingBy:, roundingQuotient:)`.
+  func divided(by divisor: Self, rounding rule: RoundingRule) -> Self
+
+  /// The remainder of dividing this value by `divisor`, using the specified
+  /// rounding `rule`.
+  ///
+  /// Equivalent to
+  /// ~~~~
+  /// let quotient = self.divided(by: divisor, rounding: rule)
+  /// return self - q*quotient
+  /// ~~~~
+  ///
+  /// A few of the available rounding rules are worth calling out in particular:
+  ///
+  /// - `.towardZero` matches the behavior of the `%` operator.
+  ///
+  /// - `.down` has the property the sign of the result always matches the
+  ///   sign of the divisor, and so when the divisor is positive, the result
+  ///   is always in `0 ..< divisor`.
+  ///
+  /// - `.toNearestOrEven` and `.toNearestOrAwayFromZero` are sometimes used
+  ///   because they produce the smallest magnitude remainder.
+  ///
+  /// - Parameters:
+  ///   - divisor: the number to divide by to get the remainder.
+  ///   - rule: the rounding rule to apply to the quotient.
+  ///
+  /// See also `mod(_:)`, `divided(by:,rounding:)`,
+  /// `quotientAndRemainder(dividingBy:, roundingQuotient:)`, and `isDivisible(by:)`.
+  func remainder(
+    dividingBy divisor: Self,
+    roundingQuotient rule: RoundingRule
+  ) -> Self
+
+  /// The mathematical modulo operation.
+  ///
+  /// This operation is similar to the `%` operator or `.remainder`, but
+  /// differs importantly in how it handles negative numbers. The `modulus`
+  /// must be positive, and the result is always non-negative, even if `self`
+  /// is negative.
+  ///
+  /// - Parameters:
+  ///   - modulus: must be positive.
+  /// - Returns: the unique integer in `0 ..< modulus` that satisfies
+  ///   `self = n*modulus + result` for some integer `n`.
+  ///
+  /// See also `remainder(dividingBy:, roundingQuotient:)` and `isDivisible(by:)`.
+  func mod(_ modulus: Self) -> Self
```

Comment:
I really like the inclusion of this and the divisible/even/odd methods. 👍

Reply:
They're right on the edge between noise and good additions for me, but I'm leaning more and more to including them, because the cost of doing so is pretty low.